consensus/ibft/ibft.go

```diff
@@ -734,9 +730,22 @@ func (i *Ibft) runAcceptState() { // start new round
 	parent := i.blockchain.Header()
 	number := parent.Number + 1
 
-	if number != i.state.view.Sequence {
-		i.logger.Error("sequence not correct", "parent", parent.Number, "sequence", i.state.view.Sequence)
-		i.setState(SyncState)
+	switch {
+	case number > i.state.view.Sequence:
+		// Node has synced already but state needs to update
+
+		i.logger.Info("next block height exceeds current sequence, unlock block & catch up", "next block height", number, "current sequence", i.state.view.Sequence)
+		if i.state.locked {
+			i.state.unlock()
+		}
+
+		i.startNewSequence(number)
+	case number < i.state.view.Sequence:
+		// For some reason, sequence exceeds current block height
+		// Needs to wait until other validators catch up
+
+		i.logger.Warn("sequence exceeds latest block height", "latest", parent.Number, "sequence", i.state.view.Sequence)
+		i.handleStateErr(errIncorrectSequence)
```

Comment:
I'm trying to figure out if this is even possible with the added verification flow.

By possible, I mean your latest block header from storage being `!=` from the next block that should be in the sequence. We update the sequence during `Commit` state and `Sync` state. If the flow is correct for both, then this can never be possible.

What do you think?

cc @dbrajovic

Reply:
I fixed and move catch up process into the end of Sync process

Reply:
You're talking about commits:
* [9756882](https://github.com/0xPolygon/polygon-edge/pull/581/commits/975688215a644c51e4873dc4c9e5f471688166e5)
* [a80dca3](https://github.com/0xPolygon/polygon-edge/pull/581/commits/a80dca3919f2e510ff097c5eab53b08d686e7ff0)

Reply:
Yes